App/BitBar/ExecutablePlugin.m

```diff
@@ -70,86 +70,97 @@ -(void)performRefreshNow:(NSMenuItem*)menuItem{
 }
 
 -(BOOL)refresh {
+  __unsafe_unretained ExecutablePlugin *weakSelf = self;
   [self.lineCycleTimer invalidate];
   self.lineCycleTimer = nil;
   [self.refreshTimer invalidate];
   self.refreshTimer = nil;
-    
+
   // execute command
   dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT,0),  ^{
-    [self refreshContentByExecutingCommand];
+    [weakSelf refreshContentByExecutingCommand];
     dispatch_sync(dispatch_get_main_queue(), ^{
-      
-      self.lastUpdated = NSDate.new;
-      
-      [self rebuildMenuForStatusItem:self.statusItem];
-      
-      // reset the current line
-      self.currentLine = -1;
-      
-      // update the status item
-      [self cycleLines];
-      
-      // sort out multi-line cycler
-      if (self.isMultiline) {
+      if (weakSelf) {
+        __strong ExecutablePlugin* strongSelf = weakSelf;
```

Comment:
Not sure that it would matter much either way given the actual execution is in the `refreshContentByExecutingCommand` call prior to the `dispatch_sync`.  It's not hurting anything to be `dispatch_sync` and I don't really have a good reason to change it.  I'd just leave it be for now and call it a day.